static/js/latest-news.js

```diff
@@ -0,0 +1,90 @@
+(function() {
+  function _slugify(string) {
+    const a = 'àáäâãåăæąçćčđďèéěėëêęğǵḧìíïîįłḿǹńňñòóöôœøṕŕřßşśšșťțùúüûǘůűūųẃẍÿýźžż·/_,:;'
+    const b = 'aaaaaaaaacccddeeeeeeegghiiiiilmnnnnooooooprrsssssttuuuuuuuuuwxyyzzz------'
+    const p = new RegExp(a.split('').join('|'), 'g')
+  
+    return string.toString().toLowerCase()
+      .replace(/\s+/g, '-') // Replace spaces with -
+      .replace(p, c => b.charAt(a.indexOf(c))) // Replace special characters
+      .replace(/&/g, '-and-') // Replace & with 'and'
+      .replace(/[^\w\-]+/g, '') // Remove all non-word characters
+      .replace(/\-\-+/g, '-') // Replace multiple - with single -
+      .replace(/^-+/, '') // Trim - from start of text
+      .replace(/-+$/, '') // Trim - from end of text
+  }
+
+  function _articleDiv(article, gtmEventLabel) {
+    const parsedDate = new Date(article.date);
+    var monthNames = [
+      "January", "February", "March",
+      "April", "May", "June", "July",
+      "August", "September", "October",
+      "November", "December"
+    ];
+    const formattedDate = parsedDate.getDate() + ' ' + monthNames[parsedDate.getMonth()] + ' ' + parsedDate.getFullYear();
+    const link = '<a href="/blog/' + article.slug + '">' + article.title.rendered + '</a>';
+    const header = "<h4>" + link + "</h4>";
+    const date = '<p class="u-no-padding--top"><em><time pubdate datetime="' + article.date + '">' + formattedDate + '</time></em></p>';
+    const div = document.createElement("div");
+    div.classList.add("col-3");
+    div.innerHTML = header + date;
+
+    div.querySelector('a').onclick = function() {
+      dataLayer.push(
+        {
+          'event': 'GAEvent',
+          'eventCategory': 'blog',
+          'eventAction': gtmEventLabel + ' news link',
+          'eventLabel': _slugify(article.title),
+          'eventValue': article.link
+        }
+      );
+    }
+
+    return div
+  }
+
+  function _latestArticlesCallback(latestNewsContainer, spotlightContainer, gtmEventLabel) {
+    return function(event) {
+      const containerForLatestNews = document.querySelector(latestNewsContainer);
+
+      // Clear out latest news container
+      while (containerForLatestNews.hasChildNodes()) {
+        containerForLatestNews.removeChild(containerForLatestNews.lastChild);
+      }
+
+      const data = JSON.parse(event.target.responseText);
+
+      const latestPinned = data.latest_pinned_articles[0];
```

Comment:
This will error if offline as the error response will not contain `latest_pinned_articles`

Reply:
If offline...

Reply:
What would you expect to happen other than an error?

Reply:
I deliberately removed the `try .. catch` statements [here](https://github.com/canonical-web-and-design/cn.ubuntu.com/blob/85057516f6f27f02325f948ceeb580fe73c20781/static/js/latest-news.js#L96) as they simply catch an error to then send a different error alert. They have no idea which error they're catching and masking.

If I was anyone - a user, a developer - and I knew about JavaScript errors, I would find the native error much more useful.

Reply:
I guess we *could* actually update the text the user sees. That might have value, but I'd like to suggest we don't bother in this first pass. Let's see how often it breaks. If it's often then we could consider giving users more info.